vraptor-core/src/main/java/br/com/caelum/vraptor/serialization/gson/GsonBuilderWrapper.java

```diff
@@ -90,9 +91,20 @@ private void registerAdapter(Class<?> adapterType, Object adapter) {
 	}
 	
 	private Class<?> getAdapterType(Object adapter) {
-		Type[] genericInterfaces = adapter.getClass().getGenericInterfaces();
-		ParameterizedType type = (ParameterizedType) genericInterfaces[0];
-		Type actualType = type.getActualTypeArguments()[0];
+		final Class<?> klazz;
+		if(isCDIProxy(adapter.getClass())){
+			final String[] split = adapter.getClass().getName().split("\\$Proxy\\$");
```

Comment:
please, use `CDIProxies.extractRawTypeIfPossible` instead.
As I can see, it could fix the problem without the `if` and reflection you've used